### ISSUE TITLE
Cover and table of content are published at the end of the e-book

### ISSUE THREAD

**Issue by WillemPeeters** [eLAIX/eLAIX#39]
Any time I export an odt file to ePub the cover and table of contents are shown at the end of my book, not at the opening.

**Comment by harry-boldt** [comment]
What is your reader oder reading software? Please check the generated epub file with readium (http://readium.org), that I use as my testing environment. If the problem still exists on readium, you can send me your odt-file and I'll take a look at it.

**Comment by lukekendall** [comment]
I have the same problem: eLAIX exports the book, then the cover, then the table of contents! I am using calibre as the reader (calibre v1.25.0 under Ubuntu 14.04).  I installed readium in Chromium and tried the .epub produced by eLAIX in that: it could display the table of contents (but gives no indication where the TOC is stored, which may make sense); but the cover page didn't seem to exist at all.  So calibre looks to me to be providing a more faithful version of the user experience.

**Comment by lukekendall** [comment]
I can send you a subset .odt that shows the problem (and the .epub too, if you like).  Would I do that by using the "Attach images by dragging & dropping or selecting them" link on the form, even though they're not images?

**Comment by harry-boldt** [comment]
No, only images are allowed here. Send your document to info@elaix.org.

**Comment by lukekendall** [comment]
On 18/04/15 18:28, Harry Boldt wrote:
> No, only images are allowed here. Send your document to info@elaix.org
> <mailto:info@elaix.org>.

Okay, I've reduced the cover image so the .odt is not too big to email 
you.  I've included the .epub it produces, too.

luke

**Comment by lukekendall** [comment]
On 18/04/15 19:50, Luke (gmail) wrote:
> On 18/04/15 18:28, Harry Boldt wrote:
>> No, only images are allowed here. Send your document to info@elaix.org
>> <mailto:info@elaix.org>.
>
> Okay, I've reduced the cover image so the .odt is not too big to email
> you.  I've included the .epub it produces, too.
>
> luke

By the way, I loaded the .epub onto my PocketBook ereader: it displayed 
a blank page, then the TOC, then the book contents; but never the cover 
image at all.

Just thought I'd let you know in case it's helpful.

Regards,

luke

**Comment by ch3lu** [comment]
I had the same problem but found a solution. Editing the .epub file with an editor like Sigil, you will find that the content.opf has a spine section and the two first items are cover and htmldoc with linear="no". Modifying linear value to yes the cover and the table of content will be shown at the beginning of the epub. I've tried it with readium, Apple iBooks and on my Kindle device (previously converted to MOBI with Calibre).

I'vent find a way to modify this parameter in the eLAIX menus, so it could be a good feature to implement in further versions.

Hope it's helpful.  

Regards,
